agent/php_environment.c

```diff
@@ -511,6 +511,124 @@ static void nr_php_get_environment_variables(TSRMLS_D) {
                    __func__, NR_PHP_PROCESS_GLOBALS(env_labels));
 }
 
+#define DOCKER_ID_V2_STRLEN (64)
+#define MAX_LINE_COUNT (1000)  // Upper bound for number of lines to read
+#define MAX_INNER_LOOP (10)    // Upper bound for number of token loops
+/*
+ * Purpose:
+ *    Extract the 64-byte hexadecimal Docker cgroup ID from
+ *     /proc/self/mountinfo
+ */
+char* nr_php_parse_v2_docker_id(const char* cgroup_fname) {
+  char* line_ptr = NULL;
+  char* token = NULL;
+  char* retval = NULL;
+  char* saveptr = NULL;
+  bool found = false;
+  int line_count = 0;
+  int token_loop_count = 0;
+  FILE* fd = NULL;
+  size_t len = 0;
+  nr_regex_t* regex = NULL;
+
+  if (NULL == cgroup_fname) {
+    return NULL;
+  }
+
+  // check if file exists
+  if (SUCCESS != access(cgroup_fname, F_OK)) {
+    nrl_verbosedebug(NRL_AGENT, "File not found: %s", cgroup_fname);
+    return NULL;
+  }
+
+  // open file
+  fd = fopen(cgroup_fname, "r");
+  if (NULL == fd) {
+    nrl_warning(NRL_AGENT, "Failed to open %s", cgroup_fname);
+    return NULL;
+  }
+
+  // compile regex to verify hexadecimal chars
+  regex = nr_regex_create("^[a-fA-F0-9]+$", 0, 0);
+
+  if (NULL == regex) {
+    nrl_error(NRL_AGENT, "Error: regex creation failed");
+    return NULL;
```

Comment:
need an `fclose(fd);` before the return.

Reply:
Fixed in 65bd871c117f5dddde7813c8e3f1aec9c04558e7